### PR TITLE
Added chucker dev extension to help debug network requests

### PR DESCRIPTION
**Pull request by chege4179**
## Checklist

- [ ] New tests were added/Existing Modified

### CODE REVIEW

**Comment by chege4179** [comment]
chucker

File: app/src/main/java/com/github/odaridavid/weatherapp/di/ClientModule.kt [hunk 1]
```diff
@@ -38,10 +42,21 @@ object ClientModule {
...
-    fun provideOkhttpClient(loggingInterceptor: HttpLoggingInterceptor): OkHttpClient =
+    fun provideOkhttpClient(
+        @ApplicationContext context:Context,
+        loggingInterceptor: HttpLoggingInterceptor
+    ): OkHttpClient =
         OkHttpClient.Builder()
             .connectTimeout(60L, TimeUnit.SECONDS)
             .addInterceptor(loggingInterceptor)
+            .addInterceptor(
+                ChuckerInterceptor.Builder(context = context)
```
Comment (jumaallan) [inline app/src/main/java/com/github/odaridavid/weatherapp/di/ClientModule.kt]: Refactor this interceptor to look like `loggingInterceptor`

Write a provide method similar to 

```kotlin
@Provides
    @Singleton
    fun provideLoggingInterceptor()
```

Then provide it here 

```kotlin
  fun provideOkhttpClient(
        @ApplicationContext context:Context,
        loggingInterceptor: HttpLoggingInterceptor
    )
```

**Review by jumaallan — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/odaridavid/weatherapp/di/ClientModule.kt (hunk 1 above)
Comment (chege4179) [reply app/src/main/java/com/github/odaridavid/weatherapp/di/ClientModule.kt]: On it

**Comment by chege4179** [comment]
@jumaallan  Made the requested changes

**Review by jumaallan — APPROVED** [review]
LGTM

**Pull request merged**